### DISCUSSION TITLE
[Question]: What is the protocol that the coordinator used to communicate with the serial port

### DISCUSSION THREAD

**Discussion by Shangzewen** [Koenkk/zigbee2mqtt#14543] [General]
Dear developers
I trying to intercept the zigbee packets by using the Zigbee dongle Model: CC2531

The method i am using is create a virtual port by using socat then intercept the communication between the coordinator and the serial port.
Right now i am able to intercept the communication but the message is encrypted, could you proved me some advice that which protocol that the coordinator used to communicate with the serial port? or is there any other way i can intercept the unencrypted zigbee packets?

Thanks for all the help

**Comment by Koenkk** [comment]
See https://github.com/Koenkk/zigbee-herdsman/blob/master/docs/Z-Stack%20Monitor%20and%20Test%20API.pdf

**Reply by Shangzewen** [reply]
Thanks Koenkk for the fast reply. On the other hand, how do you decode this ZNP message, is there any existing decoder for scapy or wireshark which can be used to decode this or you create your own dissector?

**Reply by Koenkk** [reply]
An own parser has been implemetented: https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/z-stack/unpi

**Reply by Shangzewen** [reply]
Wow, its fantastic, is there any readme which can help to under stand the code and some instruction to run the parser? Thanks again for the help

**Reply by Koenkk** [reply]
Sorry but no docs are available. I hope you are able to reverse engineer it based on the code.
